model/map-jpa/src/main/resources/META-INF/jpa-clients-changelog-1.xml

```diff
@@ -17,37 +17,55 @@ limitations under the License.
-                metadata jsonb
-            );
-
-            create index client_entityVersion on client(entityVersion);
-            create index client_realmId_clientId on client(realmId, clientId);
-            create index client_scopeMappings on client using gin ((metadata->'fScopeMappings') jsonb_path_ops);
-
-            create table client_attribute (
-                id uuid primary key not null,
-                fk_client uuid references client(id) on delete cascade,
-                name varchar(255),
-                value text
-            );
-
-            create index client_attr_fk_client on client_attribute(fk_client);
-            create index client_attr_name_value on client_attribute(name, (value::varchar(250))); 
-        </sql>
+        <createTable tableName="client">
+            <column name="id" type="UUID">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="metadata" type="json"/>
+        </createTable>
+        <ext:addGeneratedColumn tableName="client">
+            <ext:column name="entityversion" type="INTEGER" jsonColumn="metadata" jsonProperty="entityVersion"/>
+            <ext:column name="realmid" type="VARCHAR(36)" jsonColumn="metadata" jsonProperty="fRealmId"/>
+            <ext:column name="clientid" type="VARCHAR(255)" jsonColumn="metadata" jsonProperty="fClientId"/>
+            <ext:column name="protocol" type="VARCHAR(36)" jsonColumn="metadata" jsonProperty="fProtocol"/>
+            <ext:column name="enabled" type="BOOLEAN" jsonColumn="metadata" jsonProperty="fEnabled"/>
+        </ext:addGeneratedColumn>
+        <createIndex tableName="client" indexName="client_entityVersion">
+            <column name="entityversion"/>
+        </createIndex>
+        <createIndex tableName="client" indexName="client_realmId_clientId">
+            <column name="realmid"/>
+            <column name="clientid"/>
+        </createIndex>
+        <ext:createJsonIndex tableName="client" indexName="client_scopeMappings">
+            <ext:column jsonColumn="metadata" jsonProperty="fScopeMappings"/>
+        </ext:createJsonIndex>
 
+        <createTable tableName="client_attribute">
+            <column name="id" type="UUID">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="fk_client" type="UUID">
+                <constraints foreignKeyName="client_attr_fk_client_fkey" references="client(id)" deleteCascade="true"/>
+            </column>
+            <column name="name" type="VARCHAR(255)"/>
+            <column name="value" type="text"/>
+        </createTable>
+        <createIndex tableName="client_attribute" indexName="client_attr_fk_client">
+            <column name="fk_client"/>
+        </createIndex>
+        <createIndex tableName="client_attribute" indexName="client_attr_name_value">
+            <column name="name"/>
+            <column name="(value::varchar(250))"/>
```

Comment:
Can you use the same trick as in https://github.com/keycloak/keycloak/blob/6383f073c26dc2f3ca53a605a74b67ec2cee16f2/model/jpa/src/main/resources/META-INF/jpa-changelog-14.0.0.xml#L60-L79

Reply:
I'll test this